### PR TITLE
Let zipTree() and tarTree() be available on injected service

### PR DESCRIPTION
**Pull request by eskatos**
to create read-only archive FileTree instances where the Project isn't available.

See https://github.com/gradle/gradle/issues/13277

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test RFM

**Comment by eskatos** [comment]
@bot-gradle test RFM

**Comment by eskatos** [comment]
@bot-gradle test RFN

**Comment by eskatos** [comment]
@bot-gradle test RFM

File: subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java [hunk 1]
```diff
@@ -63,6 +64,26 @@
...
 
+    /**
+     * <p>Creates a read-only {@code FileTree} which contains the contents of the given ZIP file, as defined by {@link Project#zipTree(Object)}.</p>
+     *
+     * @param zipPath The ZIP file. Evaluated as per {@link Project#file(Object)}.
+     * @return the file tree. Never returns null.
+     * @since 6.6
+     */
+    @Incubating
+    FileTree zipTree(Object zipPath);
```
Comment (adammurdoch) [inline subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java:75]: This doesn't really have anything to do with the project layout (which is concerned with querying certain key locations of a project). `ProjectLayout` also isn't available outside of project scoped build logic.

Some better places might be:
- On `RegularFile` (without the parameter)
- On `Directory` to live alongside `files()`.
- On `FileSystemOperations`
- On a new service interface that concerns itself with Zip and Tar operations.

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -58,6 +58,17 @@ This improves the likelihood of [build cache hits](userguide/build_cache.html) w
 See the [userguide](userguide/more_about_tasks.html#sec:meta_inf_normalization) for further information.  Note that this API is incubating and will likely change in future releases as support 
 is expanded for normalizing properties files outside of `META-INF`.
 
+## Improvements for plugin authors
+
+### New ZIP and TAR `FileTree` factory methods
+
+Previously, it was only possible to create a `FileTree` for a ZIP or TAR archive by using the APIs provided by a `Project`.
+However, a `Project` object is not always available.
```
Comment (adammurdoch) [inline subprojects/docs/src/docs/release/notes.md:66]: If a `Project` object isn't available, then a `ProjectLayout` is also unlikely to be available.

File: subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java (hunk 1 above)
Comment (eskatos) [reply subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java:75]: Moved to their own service in https://github.com/gradle/gradle/pull/13305
Convenience on `RegularFile` or `Directory` could be added later.

**Pull request merged**